server/src/test/java/org/candlepin/dto/api/v1/CertificateTranslatorTest.java

```diff
@@ -55,6 +55,7 @@ protected CertificateTranslator initObjectTranslator() {
 
     @Override
     protected Certificate initSourceObject() {
+        //FIXME: This should not be returning an IdentityCertificate, since this class is generic.
```

Comment:
Unless the identity cert explicitly does something it shouldn't, it's fine here. So long as it's not misbehaving, any object adhering to the Certificate interface is acceptable.

Reply:
It's not about the behaviour of it, but the type. Unfortunately, when using these initSourceObject methods, we use the objects they create to populate other, parent objects, which expect a certain type. For example, when using this initSourceObject to populate an EntitlementCertificate in EntitlementTranslatorTest, I have to cast its result to EntitlementCertificate (because that is what is expected), and get "ClassCastException: IdentityCertificate cannot be cast to EntitlementCertificate". This forced me not to use this method, but to create an EntitlementCertificate manually.

Reply:
Subclasses can redeclare a method to be more specific than its parent, which is what you should do in such a case. As far as this particular method goes, returning an IdentityCertificate is still valid here, as the contract only states that it returns any Certificate.